math/uint.go

```diff
@@ -140,6 +140,11 @@ func (u Uint) MarshalJSON() ([]byte, error) {
 	return marshalJSON(u.i)
 }
 
+// MarshalYAML returns the YAML representation.
+func (i Uint) MarshalYAML() (interface{}, error) {
```

Comment:
I thought we deprecated all yaml marshalling. Why do we still need it?

Reply:
I haven't seen any announcement about that. If there was such a decision then it should be better communicated.

Moreover, community uses it in various places (including proto). Printing to screen uses yaml often (CLI).

Reply:
Custom types should not include custom MarshalJSON and MarshalYAML methods. It has caused problems in the past. We marshal yaml by going through json anyway

Reply:
hmm, I saw in many places yaml tags in proto and in implementation (mainly for user output):
+ umee uses it , (even @alexanderbez added it) 
+ osmosis uses it, cc: @ValarDragon 
+ Provenance is using it, cc: @dwedul-figure 
+ SDK is using it in `client`.

Either we communicate with communicate clearly in one of the migration notes or create a guideline.
Otherwise we get into unexpected bugs, like this one when `Uint` value was missing in user output.

Reply:
Yaml tags being in proto files is something we identified that needed to be removed a long time ago and I thought we did actually

Reply:
Furthermore I am not aware of anywhere where yaml tags or MarshalYAML methods would be used by any tools. We generate YAML in the CLI but converting from JSON to YAML. There is no custom YAML encoding supported by the SDK in anyway afaik

Reply:
See few places in this file: https://github.com/cosmos/cosmos-sdk/blob/v0.46.0/x/auth/vesting/types/vesting_account.go#L175, it goes through helper structure.
And like I said, other projects are using it  in `String()` method, eg: https://github.com/osmosis-labs/osmosis/blob/v11.0.0/x/mint/types/params.go#L101

Reply:
So I suggest we make a clear communication in one of the channels (or blog post) and maybe a guide in the SDK repo?

Reply:
Where did I add YAML for umee @robert-zaremba? Maybe it was a mistake. As pointed out, we don't need custom YAML marshalers. All output already uses YAML for human readable output (i.e. text). There's no need for these methods.

Reply:
@alexanderbez For example here: https://github.com/umee-network/umee/commit/9423f0e5#diff-6d78070081cfdf6017b58a94d19c1a45b89451b7c54f31734584c9caca207ce9R31
If `String()` method uses yaml directly, without going through JSON, then some attributes will be missing in the output, for example `Uint` (which I had to add in other place).

I'm just pinging people so we are all aligned in the ecosystem.

Reply:
Skipping JSON and going to YAML directly is the bug. There is no correct YAML renderer of protobuf types. That's why all the direct YAML stuff was deprecated. You need proto JSON first and then you need to turn that into YAML